bumble/a2dp.py

```diff
@@ -528,13 +539,113 @@ def __str__(self) -> str:
         )
 
 
+# -----------------------------------------------------------------------------
+@dataclasses.dataclass
+class OpusMediaCodecInformation(VendorSpecificMediaCodecInformation):
+    channel_mode: int
+    frame_size: int
+    sampling_frequency: int
+
+    class ChannelMode(enum.IntEnum):
+        MONO = 0
+        STEREO = 1
+        DUAL_MONO = 2
+
+    CHANNEL_MODE_BITS = {
+        ChannelMode.MONO: 1 << 0,
+        ChannelMode.STEREO: 1 << 1,
+        ChannelMode.DUAL_MONO: 1 << 2,
+    }
```

Comment:
I think these constants should be flag instead of enum, so that we don't need additional maps and `from_discrete_values` to build and serialize codec informations (also applicable to SBC and AAC).

```py

class OpusCodecInformation:
  """OPUS codec information."""

  class ChannelMode(enum.IntFlag):
    MONO = 0x01
    STEREO = 0x02
    DUAL_MONO = 0x04


  class FrameSize(enum.IntFlag):
    SIZE_10_MILLISECONDS = 0x08
    SIZE_20_MILLISECONDS = 0x10


  class SamplingRate(enum.IntFlag):
    RATE_48000 = 0x80

  sample_rate: SamplingRate
  channel_mode: ChannelMode
  frame_size: FrameSize

  VENDOR_ID: ClassVar[int] = 0xE0
  CODEC_ID: ClassVar[int] = 0x01

  def __bytes__(self) -> bytes:
    return struct.pack(
        '<IHB',
        self.VENDOR_ID,
        self.CODEC_ID,
        self.sample_rate | self.channel_mode | self.frame_size,
    )
```

Reply:
Yes, I think that's much better. I've done that for all 3 types now.
Initially, I thought it would be convenient to treat some fields like sampling frequencies as ints instead of enums, but that just complicates everything for no good reason. Using `IntFlag` for everything is the right approach.